app/src/main/java/org/fossasia/susi/ai/login/WelcomeActivity.java

```diff
@@ -0,0 +1,186 @@
+
+/**
+ * Created by Arshil on 21/08/17.
+ */
+
+public class WelcomeActivity extends AppCompatActivity {
+
+    private ViewPager mImageViewPager;
+    private int[] layouts;
+    private Button btnSkip, btnNext;
+    private PrefManager prefManager;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Checking for first time launch - before calling setContentView()
+        prefManager = new PrefManager(this);
+        if (!prefManager.isFirstTimeLaunch()) {
+            launchHomeScreen();
+            finish();
+        }
+
+
+        // Making notification bar transparent
+        if (Build.VERSION.SDK_INT >= 21) {
+            getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
+        }
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
+                WindowManager.LayoutParams.FLAG_FULLSCREEN);
+        setContentView(R.layout.activity_welcome);
+
+        mImageViewPager = (ViewPager) findViewById(R.id.pager);
+        TabLayout tabLayout = (TabLayout) findViewById(R.id.tabDots);
+        tabLayout.setupWithViewPager(mImageViewPager, true);
+
+
+
+        btnSkip = (Button) findViewById(R.id.btn_skip);
+        btnNext = (Button) findViewById(R.id.btn_next);
+
+
+        // layouts of all welcome sliders
+        layouts = new int[]{
+                R.layout.welcome_slide1,
+                R.layout.welcome_slide2,
+                R.layout.welcome_slide3,
+                R.layout.welcome_slide4};
+
+        // making notification bar transparent
+        changeStatusBarColor();
+        MyViewPagerAdapter myViewPagerAdapter = new MyViewPagerAdapter();
+        mImageViewPager.setAdapter(myViewPagerAdapter);
+        mImageViewPager.addOnPageChangeListener(viewPagerPageChangeListener);
+
+        btnSkip.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                launchHomeScreen();
+            }
+        });
+
+        btnNext.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // checking for last page
+                // if last page home screen will be launched
+                int current = getItem(+1);
+                if (current < layouts.length) {
+                    //  to next screen
+                    mImageViewPager.setCurrentItem(current);
+                } else {
+                    launchHomeScreen();
+                }
+            }
+        });
+    }
+
+
+    private int getItem(int i) {
+        return mImageViewPager.getCurrentItem() + i;
+    }
+
+    private void launchHomeScreen() {
+        prefManager.setFirstTimeLaunch(false);
+        startActivity(new Intent(WelcomeActivity.this, LoginActivity.class));
+        finish();
+    }
+
+    //  viewpager change listener
```

Comment:
Avoid obvious comments. Only add comments where necessary. Like when there is an instance when something is difficult to explain.